microservices/CFDP/app/models/cfdp_source_transaction.rb

```diff
@@ -420,32 +421,36 @@ def handle_nak(pdu_hash)
         cmd_params[@item_name] = @metadata_pdu
         cfdp_cmd(@destination_entity, @target_name, @packet_name, cmd_params, scope: ENV['OPENC3_SCOPE'])
       else
-        # Send File Data PDU(s)
-        offset = start_offset
-        source_file.seek(offset, IO::SEEK_SET)
-        while true
-          bytes_remaining = end_offset - offset
-          break if bytes_remaining <= 0
-          if bytes_remaining >= max_read_size
-            read_size = max_read_size
-          else
-            read_size = bytes_remaining
+        if source_file
+          # Send File Data PDU(s)
+          offset = start_offset
+          source_file.seek(offset, IO::SEEK_SET)
+          while true
+            bytes_remaining = end_offset - offset
+            break if bytes_remaining <= 0
+            if bytes_remaining >= max_read_size
+              read_size = max_read_size
+            else
+              read_size = bytes_remaining
+            end
+            file_data = source_file.read(read_size)
+            break if file_data.nil? or file_data.length <= 0
+            file_data_pdu = CfdpPdu.build_file_data_pdu(
+              offset: offset,
+              file_data: file_data,
+              file_size: file_size,
+              source_entity: @source_entity,
+              transaction_seq_num: @transaction_seq_num,
+              destination_entity: @destination_entity,
+              segmentation_control: @segmentation_control,
+              transmission_mode: @transmission_mode)
+            cmd_params = {}
+            cmd_params[@item_name] = file_data_pdu
+            cfdp_cmd(@destination_entity, @target_name, @packet_name, cmd_params, scope: ENV['OPENC3_SCOPE'])
+            offset += file_data.length
           end
-          file_data = source_file.read(read_size)
-          break if file_data.nil? or file_data.length <= 0
-          file_data_pdu = CfdpPdu.build_file_data_pdu(
-            offset: offset,
-            file_data: file_data,
-            file_size: file_size,
-            source_entity: @source_entity,
-            transaction_seq_num: @transaction_seq_num,
-            destination_entity: @destination_entity,
-            segmentation_control: @segmentation_control,
-            transmission_mode: @transmission_mode)
-          cmd_params = {}
-          cmd_params[@item_name] = file_data_pdu
-          cfdp_cmd(@destination_entity, @target_name, @packet_name, cmd_params, scope: ENV['OPENC3_SCOPE'])
-          offset += file_data.length
+        else
+          OpenC3::Logger.error("CFDP Cannot Handle NAK without source_file: #{@id}", scope: ENV['OPENC3_SCOPE'])
```

Comment:
Log the error but continue ... is this going to fix the original issue?

Reply:
This else case shouldn't happen.   But making sure to handle if it does with just a log message.

Reply:
The fix was handling the NAK for the metadata PDU.